payload_designer/components/diffractors.py

```diff
@@ -267,31 +279,62 @@ def get_undeviated_wavelength(self, angle_in, order=1, index_in=1, index_out=1):
         angle_1 = angle_in + self.apex_angle
         angle_2 = snell(angle=angle_1, n_1=index_in, n_2=self.index_prism)
         angle_3 = self.apex_angle - angle_2
-        angle_4 = snell(angle=angle_3, n_1=self.self.index_prism, n_2=self.index_seal)
+        angle_4 = snell(angle=angle_3, n_1=self.index_prism, n_2=self.index_seal)
         angle_5 = angle_4
         l_g = 2 * (np.sin(angle_5) / (order * self.fringe_frequency))
 
         return l_g
 
     def get_transmittance_theoretical(self):
         return NotImplementedError
-        angle_1 = a_in + a
-        angle_2 = physlib.snell_angle_2(angle_1=angle_1, n_1=n_1, n_2=n_2)
-        angle_3 = a - angle_2
-        angle_4 = physlib.snell_angle_2(angle_1=angle_3, n_1=n_2, n_2=n_3)
-        angle_5 = angle_4
-        L = 1 / v  # nm/lines
-
-        # diffraction efficiency
-        n_p = (np.sin((math.pi * n_g * d) / (l * np.cos(angle_5))) ** 2) + (
-            (1 / 2)
-            * (
-                np.sin(
-                    ((math.pi * n_g * d) * np.cos(2 * angle_5)) / (l * np.cos(angle_5))
-                )
-            )
-            ** 2
-        )  # angle_5 being close to bragg angle = more efficiency
-        n_p = n_p * eff_mat * eff_mat
+        # angle_1 = a_in + a
+        # angle_2 = physlib.snell_angle_2(angle_1=angle_1, n_1=n_1, n_2=n_2)
+        # angle_3 = a - angle_2
+        # angle_4 = physlib.snell_angle_2(angle_1=angle_3, n_1=n_2, n_2=n_3)
+        # angle_5 = angle_4
+        # L = 1 / v  # nm/lines
+
+        # # diffraction efficiency
+        # n_p = (np.sin((math.pi * n_g * d) / (l * np.cos(angle_5))) ** 2) + (
+        #     (1 / 2)
+        #     * (
+        #         np.sin(
+        #             ((math.pi * n_g * d) * np.cos(2 * angle_5)) / (l * np.cos(angle_5))
+        #         )
+        #     )
+        #     ** 2
+        # )  # angle_5 being close to bragg angle = more efficiency
+        # n_p = n_p * eff_mat * eff_mat
+
+        # return n_p
+
+
+    def get_efficiency(self, incident_angle, order=1, n_air=1):
+        """Calculates the efficiency as functon of fringe frequency for the VPH
+        Grism under the Bragg Condition.
+
+        Parameters
+        ----------
+        incident_angle : float
+            The angle of incidence of the light on the grism on the dcg layer
+        order : int, optional
+            The order of the grating. The default is 1.
+        n_air : float, optional
+            The refractive index of the air. The default is 1.
+
+        Returns
+        -------
+        float
+            The efficiency of the grism.
```

Comment:
For consistency with the rest of the project, let's use Google-style instead of Numpy-style docstrings. You can check out the format here:https://google.github.io/styleguide/pyguide.html#s3.8-comments-and-docstrings